server/src/test/java/org/elasticsearch/index/translog/TranslogTests.java

```diff
@@ -1507,6 +1510,99 @@ public void testSnapshotFromStreamInput() throws IOException {
         assertEquals(ops, readOperations);
     }
 
+    public void testSnapshotTrimmedOperations() throws Exception {
+        List<Translog.Index> operations = new ArrayList<>();
+        int translogOperations = randomIntBetween(10, 100);
+        int maxTrimmedSeqNo = translogOperations - randomIntBetween(4, 8);
+
+        final long initialPrimaryTerm = this.primaryTerm.get();
+        for (int op = 0; op < translogOperations; op++) {
+            String ascii = randomAlphaOfLengthBetween(1, 50);
+            Translog.Index operation = new Translog.Index("test", "" + op, op,
+                initialPrimaryTerm, ascii.getBytes("UTF-8"));
+            operations.add(operation);
+        }
+        Randomness.shuffle(operations);
+
+        for (Translog.Index operation : operations) {
+            translog.add(operation);
+        }
+
+        primaryTerm.incrementAndGet();
+        translog.rollGeneration();
```

Comment:
see https://github.com/elastic/elasticsearch/pull/30176#discussion_r187010569

Reply:
sorry, i thought it relates to `testRandomExceptionsOnTrimOperations`